dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/transport/StorageService.java

```diff
@@ -124,41 +124,43 @@ public void downloadPart(DataChannel channel, Part part, String objectId, File o
       public Void doWithRetry(RetryContext ctx) throws IOException {
         log.debug("Download Part URL: {}", part.getUrl());
 
-        final RequestCallback callback = new RequestCallback() {
+        final RequestCallback requestCallback = new RequestCallback() {
```

Comment:
You might consider using Java 8 lambdas here and elsewhere to cut down on a lot of noise:

```java
          HttpHeaders headers =
              dataTemplate.execute(new URI(part.getUrl()), GET,

                  request -> request.getHeaders().set(RANGE, Parts.getHttpRangeValue(part)),

                  response -> {
                    try (HashingInputStream his = new HashingInputStream(Hashing.md5(), response.getBody())) {
                      channel.readFrom(his);
                      response.getHeaders().set(ETAG, his.hash().toString());
                      return response.getHeaders();
                    }
                  });
;
```